storages/jsonl-storage/src/lib.rs

```diff
@@ -0,0 +1,420 @@
+                let table_name = path
+                    .file_stem()
+                    .map_storage_err(JsonlStorageError::FileNotFound.to_string())?
+                    .to_str()
+                    .map_storage_err(JsonlStorageError::FileNotFound.to_string())?
+                    .to_owned();
+
+                self.fetch_schema(table_name.as_str())?
+                    .map_storage_err(JsonlStorageError::TableDoesNotExist.to_string())
+            })
+            .collect::<Result<Vec<Schema>>>()
+    }
+
+    async fn fetch_data(&self, table_name: &str, target: &Key) -> Result<Option<DataRow>> {
+        let row = self.scan_data(table_name)?.find_map(|result| {
+            result
+                .map(|(key, row)| (&key == target).then_some(row))
+                .unwrap_or(None)
+        });
+
+        Ok(row)
+    }
+
+    async fn scan_data(&self, table_name: &str) -> Result<RowIter> {
+        self.scan_data(table_name)
+    }
+}
+
+fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
+where
+    P: AsRef<Path>,
+{
+    let file = File::open(filename)?;
+    Ok(io::BufReader::new(file).lines())
+}
+
+#[async_trait(?Send)]
+impl StoreMut for JsonlStorage {
+    async fn insert_schema(&mut self, schema: &Schema) -> Result<()> {
+        let data_path = self.data_path(schema.table_name.as_str());
+        File::create(data_path).map_storage_err()?;
+
+        if schema.column_defs.is_some() {
+            let schema_path = self.schema_path(schema.table_name.as_str());
+            let ddl = schema.to_ddl();
+            let mut file = File::create(schema_path).map_storage_err()?;
+            write!(file, "{ddl}").map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn delete_schema(&mut self, table_name: &str) -> Result<()> {
+        let data_path = self.data_path(table_name);
+        if data_path.exists() {
+            remove_file(data_path).map_storage_err()?;
+        }
+
+        let schema_path = self.schema_path(table_name);
+        if schema_path.exists() {
+            remove_file(schema_path).map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn append_data(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist.to_string())?;
+        let table_path = JsonlStorage::data_path(self, table_name);
+
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .open(table_path)
+            .map_storage_err()?;
+
+        let labels = schema.column_defs.unwrap_or_default();
+        let labels = labels
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<_>>();
+
+        for row in rows {
+            let json_string = match row {
+                DataRow::Vec(values) => labels
+                    .iter()
```

Comment:
Now it works without `clone`.
But not sure whether this `collect` and `iter` again way is better or not.